CONTRIBUTING.md

```diff
@@ -0,0 +1,266 @@
+4. [Bug Reports](#bug-reports)
+5. [Discussions & Suggestions](#discussions-&-suggestions)
+
+
+## Things to keep in mind
+Like a lot of other modern projects, OpenTK is written for multiple platforms and operating systems.
+Therefore, it's important to keep this in mind when contributing to the project - otherwise, it may
+make accepting your contribution much more difficult. You'll want to consider that the bug you're
+experiencing might not be present on other platforms or system configurations, or that your pull
+request doesn't take all platforms into account. Sometimes this important, sometimes it's not.
+
+OpenTK is also (as previously mentioned) a very large codebase which has seen a lot of people and a
+lot of styles over the years. This is reflected in the deeper, darker parts of the codebase where
+mixed styles, weird naming, bizarre code and eldritch sorcery abound. What may seem like a small change
+on the surface could lead you down on a path of unraveling one thread after another, and what started
+off as a simple bug fix could transform into a lot of headscratching.
+
+To make this at least somewhat easier, here's a few concrete general tips which you should stick to:
+
+* Always consider cross-platform gotchas.
+* Always work in small, iterative chunks which you can easily describe.
+* It's a good idea to open your PR early, so that you can get quick feedback. Tag it with "WIP" in the title.
+* Avoid cosmetic or visual changes, unless your contribution is strictly focused on that.
+* Don't be afraid to ask, especially before diving in. There might be someone else working on the very
+same thing already!
+* Consider how your contribution might affect other contributions. Sometimes one change will break another
+if you're not careful.
+
+In terms of these guidelines, the terminology is as follows:
+* Must: If your contribution does not follow this rule, it will not be accepted.
+* Should: If your contribution does not follow this rule, it has a lower chance of being accepted.
+* May: If your contribution does not follow this rule, it's probably not going to matter that much.
+It'd be a nice touch, though.
+
+With that in mind, check the following sections for more concrete and direct guidelines.
+
+## Setting Up
+For first-time contributors, there are a few steps that you'll need to go through in order
+to start contributing.
+
+#### 1. Get a copy of the code
+First, fork OpenTK to your own profile and clone a local copy.
+
+```bash
+$ git clone git@github.com:username/opentk.git
+$ cd opentk
+$ git remote add upstream https://github.com/opentk/opentk.git
+```
+
+#### 2. Create a working branch
+Development is done against the `develop` branch - this is where all the magic happens. Your
+changes should always be based on this branch, so in order to start working, create a new branch
+with an appropriate name and base it on `develop`.
+
+```bash
+$ git checkout -b my-branch -t origin/develop
+```
+
+#### 3. Let git know who you are
```

Comment:
Maybe use this link: https://help.github.com/articles/set-up-git/ and move in on 1 position?